scipy/signal/_signaltools.py

```diff
@@ -3625,13 +3726,19 @@ def detrend(data: np.ndarray, axis: int = -1,
     """
     if type not in ['linear', 'l', 'constant', 'c']:
         raise ValueError("Trend type must be 'linear' or 'constant'.")
+
+    if isinstance(bp, int):
+       xp = array_namespace(data)
+    else:
+       xp = array_namespace(data, bp)
```

Comment:
could you add a comment linking to https://github.com/data-apis/array-api-compat/pull/147?